rvc/lib/predictors/RMVPE.py

```diff
@@ -408,15 +409,30 @@ def forward(self, audio, keyshift=0, speed=1, center=True):
             self.hann_window[keyshift_key] = torch.hann_window(win_length_new).to(
                 audio.device
             )
-        fft = torch.stft(
-            audio,
-            n_fft=n_fft_new,
-            hop_length=hop_length_new,
-            win_length=win_length_new,
-            window=self.hann_window[keyshift_key],
-            center=center,
-            return_complex=True,
-        )
+        # Zluda fall-back to CPU for FFTs since HIP SDK has no cuFFT alternative
+        if audio.device.type == "cuda" and torch.cuda.get_device_name(audio.device.index).endswith("[ZLUDA]"):
+            fft = torch.stft(
```

Comment:
Simply move `hann_window` and `audio` to CPU before using torch.stft. Like:
```py
source_device = audio.device
if audio.device.type == "cuda" and torch.cuda.get_device_name(audio.device.index).endswith("[ZLUDA]"):
  audio = audio.to('cpu')
  self.hann_window[keyshift_key] = self.hann_window[keyshift_key].to('cpu')
fft = torch.stft(
    audio,
    n_fft=n_fft_new,
    hop_length=hop_length_new,
    win_length=win_length_new,
    window=self.hann_window[keyshift_key],
    center=center,
    return_complex=True,
).to(source_device) # no-op if same device
```

Or use backend-agnostic implementation of STFT.
https://github.com/RVC-Project/Retrieval-based-Voice-Conversion-WebUI/blob/main/infer/lib/rmvpe.py#L29-L156

Reply:
> Simply move `hann_window` and `audio` to CPU before using torch.stft. Like:
> 
> ```python
> source_device = audio.device
> if audio.device.type == "cuda" and torch.cuda.get_device_name(audio.device.index).endswith("[ZLUDA]"):
>   audio = audio.to('cpu')
>   self.hann_window[keyshift_key] = self.hann_window[keyshift_key].to('cpu')
> fft = torch.stft(
>     audio,
>     n_fft=n_fft_new,
>     hop_length=hop_length_new,
>     win_length=win_length_new,
>     window=self.hann_window[keyshift_key],
>     center=center,
>     return_complex=True,
> ).to(source_device) # no-op if same device
> ```
> 
> Or use backend-agnostic implementation of STFT. https://github.com/RVC-Project/Retrieval-based-Voice-Conversion-WebUI/blob/main/infer/lib/rmvpe.py#L29-L156

this change results in

  File "G:\GitNew\Applio-Zluda\rvc\train\mel_processing.py", line 85, in spectrogram_torch
    spec = torch.stft(
  File "G:\GitNew\Applio-Zluda\env\lib\site-packages\torch\functional.py", line 650, in stft
    return _VF.stft(input, n_fft, hop_length, win_length, window,  # type: ignore[attr-defined]
TypeError: stft() received an invalid combination of arguments - got (tuple, int, int, int, Tensor, bool, bool, bool), but expected one of:
 * (Tensor input, int n_fft, int hop_length, int win_length, Tensor window, bool center, str pad_mode, bool normalized, bool onesided, bool return_complex)
 * (Tensor input, int n_fft, int hop_length, int win_length, Tensor window, bool normalized, bool onesided, bool return_complex)
      didn't match because some of the arguments have invalid types: (tuple of (Tensor,), int, int, int, Tensor, bool, bool, bool)

Reply:
`audio` became a tuple of Tensor? 🤔 Looks strange, how exactly did you apply the change?

Reply:
> `audio` became a tuple of Tensor? 🤔 Looks strange, how exactly did you apply the change?

sorry, made a typo appying the new changes, all good